engine/src/main/java/org/apache/apex/engine/plugin/loaders/ChainedPluginLocator.java

```diff
@@ -18,40 +18,40 @@
  */
 package org.apache.apex.engine.plugin.loaders;
 
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.List;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import org.apache.apex.api.plugin.Plugin;
 import org.apache.apex.engine.api.plugin.PluginLocator;
 import org.apache.hadoop.conf.Configuration;
 
 /**
  * @since 3.6.0
  */
-public class ChainedPluginLocator<T> implements PluginLocator<T>
+public class ChainedPluginLocator<T extends Plugin> implements PluginLocator<T>
```

Comment:
Is this more for readability as the type is already bounded by the PluginLocator